### ISSUE TITLE
/usr/bin/env: ‘node’: No such file or directory

### ISSUE THREAD

**Issue by amlcodes** [microsoft/vscode-remote-release#3393]
- VSCode Version: 1.48.0-insider
- Local OS Version:10.15.5 (19F101)
- Remote OS Version: ubuntu 18.04
- Remote Extension/Connection Type: Docker

When I try to run one of my debugs, I get an error
`/usr/bin/env: ‘node’: No such file or directory`

 this one in particular for debugging babel-node in development. I also get an error for the debug prod configuration. 

Steps to Reproduce:
package.json 
```
  "devDependencies": {
....
    "@babel/node": "^7.8.7",
},
"babel": {
    "env": {
      "test": {
        "presets": [
          "@babel/preset-env"
        ],
        "plugins": [
          "@babel/plugin-proposal-optional-chaining",
          "@babel/plugin-transform-runtime",
          "@babel/plugin-transform-modules-commonjs",
          "@babel/plugin-proposal-class-properties"
        ],
        "sourceMaps": "inline",
        "retainLines": true
      },
      "production": {
        "presets": [
          "@babel/preset-env"
        ],
        "plugins": [
          "@babel/plugin-proposal-optional-chaining",
          "@babel/plugin-transform-runtime",
          "@babel/plugin-transform-modules-commonjs",
          "@babel/plugin-proposal-class-properties"
        ]
      },
      "development": {
        "presets": [
          "@babel/preset-env"
        ],
        "plugins": [
          "@babel/plugin-proposal-optional-chaining",
          "@babel/plugin-transform-runtime",
          "@babel/plugin-transform-modules-commonjs",
          "@babel/plugin-proposal-class-properties"
        ],
        "sourceMaps": "inline",
        "retainLines": true
      }
    }
  },
```
launch.json
```
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "diagnosticLogging": true,
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Launch Prod",
      "skipFiles": ["<node_internals>/**"],
      "program": "${workspaceFolder}/build/index.js",
      "runtimeExecutable": "/usr/local/nvm/versions/node/v10.18.1/bin/node"
    },
    {
      "name": "Develop",
      "type": "node",
      "request": "launch",
      "protocol": "inspector",
      "program": "${workspaceRoot}/src/index.js",
      "stopOnEntry": false,
      "cwd": "${workspaceRoot}",
      // "skipFiles": ["<node_internals>/**"],
      "preLaunchTask": null,
      "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/babel-node",
      // "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/nodemon",
      "args": ["${workspaceRoot}/src/index.js"],
      "runtimeArgs": ["--nolazy", "--require", "@babel/register"],
      "env": {
        "NODE_ENV": "development"
      },
      "console": "internalConsole",
      "sourceMaps": true,
      "outFiles": []
    },
    {
      "name": "Debug Jest Tests",
      "type": "node",
      "request": "launch",
      "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/babel-node",
      "runtimeArgs": [
        "--inspect-brk",
        "${workspaceRoot}/node_modules/.bin/jest",
        "--runInBand"
      ],
      "console": "internalConsole",
      "internalConsoleOptions": "neverOpen",
      "port": 9229
    }
  ]
}
```
Dockerfile
```
FROM {some cpp base}
WORKDIR $HOME
ARG USERNAME=ubuntu
ENV NVM_DIR=/usr/local/nvm
ENV HOME /home/ubuntu
ENV PATH=${PATH}:/usr/local/share/npm-global/bin

RUN  apt-get update && apt-get -y install curl zsh git 
RUN zsh -c "$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)" ||true

RUN echo 'eval "$(starship init zsh)"' >> /home/ubuntu/.zshrc
RUN echo 'eval "$(ssh-agent -s)"' >> /home/ubuntu/.zshrc
RUN echo '. /usr/local/nvm/nvm.sh' >> /home/ubuntu/.zshrc
```

**Comment by amlcodes** [comment]
When I $PATH in my container i get 
```
zsh: no such file or directory: /usr/local/nvm/versions/node/v10.18.1/bin:/home/ubuntu/.vscode-server-insiders/bin/ff6b1c5984a4c2116a06318f07e871a4169fdb1b/bin:/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by chrmarti** [comment]
There is a fix in Remote-Containers 0.131.0 currently available with VS Code Insiders (https://code.visualstudio.com/insiders/). With that you can add `"userEnvProbe": "interactiveShell"` to your devcontainer.json to source .zshrc. Make sure zsh is the user's default shell (using `chsh` to update `/etc/passwd`).

Fixed with https://github.com/microsoft/vscode-remote-release/issues/3224.
